handler/src/main/java/io/netty/handler/ssl/util/FingerprintTrustManagerFactory.java

```diff
@@ -207,4 +252,50 @@ protected void engineInit(ManagerFactoryParameters managerFactoryParameters) thr
     protected TrustManager[] engineGetTrustManagers() {
         return new TrustManager[] { tm };
     }
+
+    /**
+     * A builder for creating {@link FingerprintTrustManagerFactory}.
+     */
+    public static final class Builder {
+
+        /**
+         * A hash algorithm for fingerprints.
+         */
+        private final String algorithm;
+
+        /**
+         * A list of fingerprints.
+         */
+        private final List<String> fingerprints = new ArrayList<String>();
+
+        /**
+         * Creates a builder.
+         *
+         * @param algorithm a hash algorithm
+         */
+        private Builder(String algorithm) {
+            this.algorithm = ObjectUtil.checkNotNull(algorithm, "algorithm");
+        }
+
+        /**
+         * Adds a fingerprint.
+         *
+         * @param fingerprint a fingerprint
+         * @return the same builder
+         */
+        public Builder fingerprint(String fingerprint) {
```

Comment:
Add method to add a List of `fingerprints`.

Reply:
honestly this feels a bit like overkill.. Let's just keep it or use `String...`

Reply:
Calling `public Builder fingerprint(String fingerprint)` every time is a little painful that's why we need something to add fingerprints in bulk. Both `List` and `String...` will do the job.

Reply:
I disagree... what is the point of a builder if you will never call its methods multiple times ?

Reply:
That's a thing too.

Reply:
I think both points are valid. The builder may be used in multiple ways. For example, someone may want to hardcode fingerprints like the following:

```java
FingerprintTrustManagerFactory f =
  FingerprintTrustManagerFactory
    .builder("SHA256")
    .fingerprint("deadbeef...")
    .fingerprint("cafecafe...")
    .build();
```

But someone may want to load fingerprints from a file. In this case, `fingerprint()` would need to be called in a loop:

```java
FingerprintTrustManagerFactory.Builder b = FingerprintTrustManagerFactory.builder("SHA256");
for (String s : loadFingerprintsFrom(file)) {
    b.fingerprint(s);
}
```

Here `fingerprint(String...)` might make it look a bit nicer:

```java
FingerprintTrustManagerFactory f =
  FingerprintTrustManagerFactory
    .builder("SHA256")
    .fingerprint(loadFingerprintsFrom(file))
    .build();
```

Anyway, I am fine to keep it as-is, or updating `fingerprint()` to accept `String...`.

Reply:
I am ok with `String...` but then call it fingerprints(...)